packages/nextjs/utils/scaffold-eth/networks.ts

```diff
@@ -56,26 +56,32 @@ export const NETWORKS_EXTRA_DATA: Record<string, TChainAttributes> = {
  * @dev returns empty string if the network is localChain
  */
 export function getBlockExplorerTxLink(network: Network, txnHash: string) {
-  const { name, chainId } = network;
+  let blockExplorerTxURL = "";
+  const { chainId } = network;
 
-  if (chainId === 31337 || chainId === 1337) {
-    // If its localChain then return empty sting
+  const chainNames = Object.keys(chains);
+
+  const targetChainArr = chainNames.filter(chainName => {
+    const wagmiChain = chains[chainName as keyof typeof chains];
+    return wagmiChain.id === chainId;
+  });
+
```

Comment:
An idea (without testing it :angel: ).

What if we make the argument `network` to be wagmi's `Chain`? So we don't need to do this conversion.

In the transactor, we could get the connected network with wagmi's `useNetwork()` (and maybe throw an error / return if the signer is on a different network)

```tsx
// useTransactor
const { chain } = useNetwork();
// check that network.chainId === chain.id => if not, error / return
getBlockExplorerTxLink(chain, ...)
```

Since we are using wagmi's Chains as source of truth, maybe we should rely on them whenever we can? Instead of using internal ethers/providers.

Reply:
Yup makes sense, lol even I was a bit uncomfortable while writing the conversion, 

> What if we make the argument network to be wagmi's Chain? So we don't need to do this conversion.

This is a great idea but the problem is `useTransactor` accepts `custom signer` too and the reason we are relying on the signers `network` is that : 
NITPICK: There might a case where people pass in the custom signer without connecting their wallet and in that case `useNetwork` and signer `network` will be different and might cause some inconsistency, we are using similar pattern in `Faucet.jsx`. 

Again the above NITPICK is a very rare case. 

Will try exploring some more alternatives 🙌

Reply:
Hey Shiv, good point!!

After pulling the threads on this, some notes:
- The only instance where we are passing a custom signer, is for the hardhat faucet (where we want to send funds from a hardhat account with funds). Can we think about other use case? Since we are in the front-end, 99.99% of the times we'll want to use the connected signer (besides the faucet, and maybe a exotic build where we do deterministic private key derivation or something)
- `getBlockExplorerTxLink` doesn't return anything for the local chain (the only instance where we are using a custom signer for the transactor :thinking: )
- `getLocalProvider` is only used in the Faucet stuff. Could be an unneeded early abstraction.
- Maybe an util like `getChainFromChainId` that returns a wagmi Chain for a given Id?

In any case, no need to make any updates rn!! just thinking out loud and pointing out some possible inconsistencies. Maybe we made some past decisions that could use some refactors for the sake of consistency / simplicity / better DX.

Thanks!

Reply:
Tysm everything makes sense!!

> Maybe we made some past decisions that could use some refactors for the sake of consistency / simplicity / better DX

+++